From 516030f287f65ca8bdab92c979ba2d328ee40506 Mon Sep 17 00:00:00 2001
From: James Hilliard <james.hilliard1@gmail.com>
Date: Fri, 31 May 2024 12:51:01 -0600
Subject: [PATCH] src/execstack.c: fix prelink_path type
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:
execstack.c: In function ‘execstack_make_rdwr’:
execstack.c:127:17: error: passing argument 1 of ‘asprintf’ from incompatible pointer type [-Wincompatible-pointer-types]
  127 |       asprintf (&prelink_path, "%s/%s", dirname, PRELINK_PROG EXEEXT);
      |                 ^~~~~~~~~~~~~
      |                 |
      |                 const char **
In file included from execstack.c:10:
/usr/include/stdio.h:403:40: note: expected ‘char ** restrict’ but argument is of type ‘const char **’
  403 | extern int asprintf (char **__restrict __ptr,
      |                      ~~~~~~~~~~~~~~~~~~^~~~~

Signed-off-by: James Hilliard <james.hilliard1@gmail.com>
Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>
Upstream: https://git.yoctoproject.org/prelink-cross/commit/?id=ff2561c02ade96c5d4d56ddd4e27ff064840a176
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 src/execstack.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/execstack.c b/src/execstack.c
index dda6bc7..5413fa2 100644
--- a/src/execstack.c
+++ b/src/execstack.c
@@ -52,7 +52,7 @@ const char *program_path;
 
 /* The full pathname of the prelink tool, or NULL if it hasn't been
    computed yet.  */
-const char *prelink_path;
+char *prelink_path;
 
 static error_t
 parse_opt (int key, char *arg, struct argp_state *state)
-- 
2.45.2

